cinema/views.py

```diff
@@ -2,0 +1,63 @@
+from rest_framework import viewsets
+
+from cinema.models import Movie, CinemaHall, Genre, Actor, MovieSession
+from cinema.serializers import (
+    MovieSerializer,
+    CinemaHallSerializer,
+    GenreSerializer,
+    ActorSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieListSerializer,
+    MovieRetrieveSerializer,
+    MovieSessionRetrieveSerializer
+)
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+        return MovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            return queryset.prefetch_related("genres", "actors")
+        return queryset
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
```

Comment:
The use of `.all()` before `.select_related()` is redundant. You can directly use `MovieSession.objects.select_related()` in the `get_queryset` method to optimize performance.